### ISSUE TITLE
Newsletters error

### ISSUE THREAD

**Issue by TheTechromancer** [blacklanternsecurity/bbot#1138]
@stryker2k2 please look at this when you get a chance. Discovered in https://github.com/blacklanternsecurity/bbot/issues/1137:

![image](https://github.com/blacklanternsecurity/bbot/assets/20261699/243bfa61-7ff5-468b-8f2d-dc81149ad251)

**Comment by stryker2k2** [comment]
@TheTechromancer - is this still needed to be looked at?

I just ran the same command string that he did and I couldn't replicate it. Then again, I don't have the API Keys loaded.

**Comment by TheTechromancer** [comment]
The problem is that the beautifulsoup helper sometimes returns False, but that case isn't handled in the module. We need to make sure that case is being handled wherever the helper is used.

![image](https://github.com/blacklanternsecurity/bbot/assets/20261699/0285998a-7870-405d-be64-6e1aefbf14f9)

**Comment by stryker2k2** [comment]
Roger that. I see what you're seeing. I'll work on it.

**Comment by Sh4d0wHunt3rX** [comment]
On bbot v1.1.7.3052rc
```
2024-03-13 10:36:21,870 [ERROR] bbot.scanner scanner.py:1062 Error in newsletters.handle_event(HTTP_RESPONSE("{'url': 'https://apf-api.eng.vn.cloud.tesla.com:443', 'timestamp': '2024-03-13T1...", module=httpx, tags={'in-scope', 'cloud-amazon', 'status-200', 'dir', 'ip-18-239-83-61'})): /root/.local/pipx/venvs/bbot/lib/python3.10/site-packages/bbot/modules/newsletters.py:36:handle_event(): 'body'
2024-03-13 10:36:21,870 [TRACE] bbot.scanner scanner.py:1063 Traceback (most recent call last):
  File "/root/.local/pipx/venvs/bbot/lib/python3.10/site-packages/bbot/scanner/scanner.py", line 1045, in _acatch
    yield
  File "/root/.local/pipx/venvs/bbot/lib/python3.10/site-packages/bbot/modules/base.py", line 627, in _worker
    await handle_event_task
  File "/root/.local/pipx/venvs/bbot/lib/python3.10/site-packages/bbot/modules/newsletters.py", line 36, in handle_event
    soup = self.helpers.beautifulsoup(event.data["body"], "html.parser")
KeyError: 'body'
```

**Comment by stryker2k2** [comment]
Wow, well... yes. It seems that `https://apf-api.eng.vn.cloud.tesla.com:443` doesn't have a body. And BBOT Newsletters should be prepared to take that into consideration.

I'll take a look into it.

**Comment by stryker2k2** [comment]
This has been fixed in the Pull Request #1154 .
